### ISSUE TITLE
headless mode not working on coles

### ISSUE THREAD

**Issue by ramzitra** [ultrafunkamsterdam/undetected-chromedriver#222]
Hello, I noticed this issue :
on headless mode , it does not work on : https://shop.coles.com.au/a/national/home
last week it was working fine with headless mode.
I tested both undetected-chromedriver v1 and v2 and both not working in headless. 
Without headless mode, it works fine and it gives the good HTML response, but in headless mode, it is giving strange HTML when we check the outer html with : driver.find_element_by_xpath('//*').get_attribute('outerHTML')

Any idea why ?
Thanks in advance

**Comment by ramzitra** [comment]
this is the line of codes :

import undetected_chromedriver as uc
import time

options = uc.ChromeOptions()
options.add_argument("--log-level=3")
# options.headless=True
# options.add_argument('--headless')

driver = uc.Chrome(options=options)

url  = "https://shop.coles.com.au/a/national/everything/browse"

driver.get(url)

time.sleep(10)

html        = driver.find_element_by_xpath('//*').get_attribute('outerHTML')
print(html)

**Comment by KOOPAKO** [comment]
try this instead https://stackoverflow.com/a/60543068 works a bit better for me when it comes to headless browsers

**Comment by ultrafunkamsterdam** [comment]
@ramzitra you are also adding double arguments. Use ether:
 ```
options.headless = True
#or 
options.add_argument('--headless') 
#or
uc.Chrome(headless=True)
```

Not combined

**Comment by ramzitra** [comment]
> @ramzitra you are also adding double arguments. Use ether:
> 
> ```
> options.headless = True
> #or 
> options.add_argument('--headless') 
> #or
> uc.Chrome(headless=True)
> ```
> 
> Not combined
> 
> > try this instead https://stackoverflow.com/a/60543068 works a bit better for me when it comes to headless browsers
> 
> = true

Hi, i already tried that and it is the same issue. It was working with undetected chromedriver headless for months but from this week the headless not working and it gives strange html response.

**Comment by ramzitra** [comment]
> try this instead https://stackoverflow.com/a/60543068 works a bit better for me when it comes to headless browsers

Yes this true as it works without --headless argument but this is only for linux , but not for Windows as xvfb does not support it.

**Comment by KOOPAKO** [comment]
> > try this instead https://stackoverflow.com/a/60543068 works a bit better for me when it comes to headless browsers
> 
> Yes this true as it works without --headless argument but this is only for linux , but not for Windows as xvfb does not support it.

Ohh I did not even realise that, my bad
